solutions/ioi-11-race.mdx

```diff
@@ -0,0 +1,84 @@
+---
+id: ioi-11-race
+source: IOI 2011
+title: Race
+author: Timothy Gao
+---
+
+[Official Analysis with Centroid Decomposition Downloaded Here](https://ioinformatics.org/page/ioi-2011/37)
+
+## Alternative Solution: Small to Large Merging
+
+Like many centroid decomposition problems, race is also solvable via the small-to-large merging technique. At each node, consider storing a map. Naively, storing pairs of Length : #_of_paths does not work, because when we move from a node to it's ancestor, all values in the map stored that node much be updated. To account for this, we can instead define all paths as global to the root. In other words, we store pairs of Length_from_root : height. This way, when we merge a smaller map to a larger one at node $u$, we can process the answer with the formulation 
+
+$\textit{min(answer, smallmapheight + largemap[K-smallmaplength] - 2*height[u])}$
+
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N\log^2N)$
+
+
+```cpp
+#include <bits/stdc++.h>
+ 
+using namespace std;
+ 
+typedef long long ll;
+typedef pair<ll, ll> pii;
+ 
+#define pb push_back
+ 
+#define f first
+#define s second
+ 
+const int maxn = 2e5+20;
+vector<pii> adj[maxn];
+map<ll, ll> info[maxn]; 
+ll dist[maxn], height[maxn]; //define dist relative to root
+int N, K, u, v; ll ret;
+ 
+void dfs1(int u, int p, ll c, int h){
+    info[u][c] = h; 
+    dist[u] = c; height[u] = h;
+    for(auto n : adj[u]){
+        if(p == n.f) continue;
+        dfs1(n.f, u, c + n.s, h + 1);
+    }
+}
+ 
+void dfs(int u, int p){
+    ll target = K + 2*dist[u];
+    for(auto n : adj[u]){
+        if(n.f == p) continue;
+        dfs(n.f, u);
+        if(info[n.f].size() > info[u].size()){
```

Comment:
throughout your code you don't add a space like I think it looks better like
if (...) { ... with a space between f and ( and a space between ) and {

not a requirement though I think but almost all of the other solutions follow this convention